File: valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        
        if (s.equals(null) || s.length() == 0)
            return false;
        
        Stack<Character> parentheses = new Stack<>();
        char[] chArr = s.toCharArray();
        
        for (char ch : chArr) {
            if (ch == '(')
                parentheses.push(')');
            else if (ch == '[')
                parentheses.push(']');
            else if (ch == '{')
                parentheses.push('}');
            else if (parentheses.empty() || parentheses.pop() != ch)
                return false;
        }
        
        return parentheses.empty();
    }
}

File: valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        parenthesis = {")":"(", "}":"{", "]":"["}
        
        for p in s:
            if p in parenthesis:
                if stack and stack[-1] == parenthesis[p]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(p)
                
        return True if not stack else False